File: aula03-managed-point/Ponto.cs
using System;

public class Ponto{

	// readonly <=> Java final
	public readonly int _z, _x, _y;

	public Ponto(int x, int y)
	{
		this._x = x;
		this._y = y;
		this._z = 0;
	}

	public double GetModule() {	 
		return Math.Sqrt((double)_x*_x + _y*_y + _z*_z);
	}

	public void Print(){
		Console.WriteLine("Version 2 Point (x = {0}, y = {1}), z = {2})", _x, _y, _z);
	}

}

File: aula02-unmanaged-dynamic-link/PontoDll/Ponto.cpp
// Ponto.cpp : Defines the exported functions by Ponto.dll
//

#include "Ponto.h"
#include <math.h>
#include <iostream>

/*
 * This is the constructor of a class that has been exported.
 * see Ponto.h for the class definition
 */
Ponto::Ponto(int x, int y){
	this->_x = x;
	this->_y = y;
}

double Ponto::getModule() {	 
	return sqrt((double)_x*_x + _y*_y);
}

void Ponto::print(char * label){
	printf("v2 %s: (x = %d, y = %d)\n", label, _x, _y);
}

